packages/webamp/js/components/Vis.tsx

```diff
@@ -0,0 +1,234 @@
+import React, {
+  useMemo,
+  useCallback,
+  useState,
+  useLayoutEffect,
+  useEffect,
+} from "react";
+
+import * as Actions from "../actionCreators";
+import * as Selectors from "../selectors";
+import { FFT } from "./FFTNullsoft";
+import { useTypedSelector, useActionCreator } from "../hooks";
+// import { usePaintOscilloscopeFrame } from "./useOscilloscopeVisualizer";
+// import { usePaintBarFrame, usePaintBar } from "./useBarVisualizer";
+import { VISUALIZERS, MEDIA_STATUS } from "../constants";
+
+import {
+  Vis as IVis,
+  VisPaintHandler,
+  BarPaintHandler,
+  WavePaintHandler,
+  NoVisualizerHandler,
+} from "./VisPainter";
+
+type Props = {
+  analyser: AnalyserNode;
+};
+
+export let PIXEL_DENSITY = 1;
+
+const fft = new FFT();
+const samplesIn = 1024; // Example input size
+const samplesOut = 512; // Example output size
+export let renderWidth: number;
+export let renderHeight: number;
+export let windowShade: boolean | undefined;
+export let doubled: boolean | undefined;
```

Comment:
What's the deal with these exported values? Used for debugging maybe? Can we remove them now?

Reply:
They're used in various places in VisPainter so that it knows the current state of the Main Window, I think I can get rid of renderWidth and renderHeight, I think "doubled" is used somewhere but I can't remember without checking, windowShade is integral to ensure the visualizer is squished down to the size expected.

Although to be real, I don't know how else I could get these values into VisPainter.ts, so `export` was the sort of temporary bandaid solution.